### PR TITLE
[WIP] CFE-3046: Alternative approach

### PR DESCRIPTION
**Pull request by oleorhagen**
More coming

Changelog: None

Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

File: libenv/unix_iface.c [hunk 1]
```diff
@@ -165,8 +165,14 @@ static void GetMacAddress(EvalContext *ctx, ARG_UNUSED int fd, struct ifreq *ifr
              (unsigned char) ifr->ifr_hwaddr.sa_data[5]);
 
     EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, name, hw_mac, CF_DATA_TYPE_STRING, "source=agent");
-    RlistAppend(hardware, hw_mac, RVAL_TYPE_SCALAR);
-    RlistAppend(interfaces, ifp->ifr_name, RVAL_TYPE_SCALAR);
+    if (!RlistContainsString(*hardware, hw_mac))
```
Comment (oleorhagen) [inline libenv/unix_iface.c]: @olehermanse took another approach second time around.

**Comment by olehermanse** [comment]
@cf-bottom jenkins with exotics, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4444/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4444/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4444/

**Comment by oleorhagen** [comment]
Suck it AIX!

**Pull request closed without merging**